packages/apollo/src/commands/service/check.ts

```diff
@@ -253,112 +260,198 @@ export default class ServiceCheck extends ProjectCommand {
-                   * variant: like a tag. prod/staging/etc
-                   * name: implementing service name inside of the graph
-                   * sha: git commit hash/docker id. placeholder for now
-                   */
-                  task.output = "Creating composed schema against the graph";
-                  const {
-                    errors,
-                    warnings,
-                    compositionValidationDetails
-                  } = await project.engine.checkPartialSchema({
-                    id: configName,
-                    graphVariant: tag,
-                    implementingServiceName: flags.serviceName || info.name,
-                    partialSchema: {
-                      sdl: info.sdl
+                const info = await project.resolveFederationInfo();
+                if (!info.sdl) {
+                  throw new Error("No SDL found for federated service");
+                }
+
+                task.output = `Attempting to compose graph with ${chalk.blue(
+                  serviceName
+                )} service's partial schema`;
+
+                const {
+                  errors,
+                  warnings,
+                  compositionValidationDetails
+                } = await project.engine.checkPartialSchema({
+                  id: configName,
+                  graphVariant: tag,
+                  implementingServiceName: serviceName,
+                  partialSchema: {
+                    sdl: info.sdl
+                  }
+                });
+
+                if (
+                  compositionValidationDetails &&
+                  compositionValidationDetails.schemaHash
+                ) {
+                  ctx.federationSchemaHash =
+                    compositionValidationDetails.schemaHash;
+                }
+
+                task.title = `Found ${pluralize(
+                  errors.length,
+                  "graph composition error"
+                )} for service ${chalk.blue(serviceName)} on graph ${chalk.blue(
+                  configName
+                )}`;
+
+                if (errors.length > 0) {
+                  const decodedErrors = errors.filter(Boolean).map(error => {
+                    // We `filter` by `Boolean` above; this will never happen and is just here to appease
+                    // TypeScript.
+                    if (!error) {
+                      return {};
+                    }
+
+                    const match = error.message.match(
+                      /^\[([^\[]+)\]\s+(\S+)\ ->\ (.+)/
+                    );
+
+                    if (!match) {
+                      // If we can't match the errors, that means they're in a format we don't recognize.
+                      // Report the entire string as the user will see the raw message.
+                      return { message: error.message };
                     }
+
+                    const [, service, field, message] = match;
```

Comment:
I'm not sure what the first value in the array is. Can we add a comment to describe what we're ignoring?

Reply:
You got it!